python/lsst/dax/apdb/cassandra/config.py

```diff
@@ -0,0 +1,218 @@
+    )
+
+    read_concurrency: int = Field(
+        default=500,
+        description="Concurrency level for read operations.",
+    )
+
+    protocol_version: int = Field(
+        default=cassandra.ProtocolVersion.V4 if CASSANDRA_IMPORTED else 4,
+        description="Cassandra protocol version to use, default is V4.",
+    )
+
+    extra_parameters: dict[str, Any] = Field(
+        default={}, description="Additional keyword parameters passed to connect() method verbatim."
+    )
+
+
+class ApdbCassandraPartitioningConfig(BaseModel):
+    """Partitioning configuration for Cassandra APDB."""
+
+    part_pixelization: str = Field(
+        default="mq3c",
+        description="Pixelization used for partitioning index.",
+    )
+
+    part_pix_level: int = Field(
+        default=11,
+        description="Pixelization level used for partitioning index.",
+    )
+
+    part_pix_max_ranges: int = Field(
+        default=128,
+        description="Max number of ranges in pixelization envelope",
+    )
+
+    time_partition_tables: bool = Field(
+        default=False,
+        description="Use per-partition tables for sources instead of partitioning by time",
+    )
+
+    time_partition_days: int = Field(
+        default=30,
+        description=(
+            "Time partitioning granularity in days, this value must not be changed after database is "
+            "initialized"
+        ),
+    )
+
+    time_partition_start: str = Field(
+        default="2018-12-01T00:00:00",
+        description=(
+            "Starting time for per-partition tables, in yyyy-mm-ddThh:mm:ss format, in TAI. "
+            "This is used only when time_partition_tables is True."
+        ),
+    )
+
+    time_partition_end: str = Field(
+        default="2030-01-01T00:00:00",
+        description=(
+            "Ending time for per-partition tables, in yyyy-mm-ddThh:mm:ss format, in TAI. "
+            "This is used only when time_partition_tables is True."
+        ),
+    )
+
+    query_per_time_part: bool = Field(
+        default=False,
+        description=(
+            "If True then build separate query for each time partition, otherwise build one single query. "
+            "This is only used when time_partition_tables is False in schema config."
+        ),
+    )
+
+    query_per_spatial_part: bool = Field(
+        default=False,
+        description="If True then build one query per spatial partition, otherwise build single query.",
+    )
+
+    @field_validator("part_pixelization")
+    @classmethod
+    def check_pixelization(cls, v: str) -> str:
+        allowed = {"htm", "q3c", "mq3c"}
+        if v not in allowed:
+            raise ValueError(f"Unexpected value for part_pixelization: {v}, allowed values: {allowed}")
+        return v
+
+
+class ApdbCassandraConfig(ApdbConfig):
+    """Configuration class for Cassandra-based APDB implementation."""
+
+    _implementation_type: ClassVar[str] = "cassandra"
+
+    contact_points: list[str] = Field(
```

Comment:
General comment about `list[str]` vs `tuple[str, ...]`. Are people editing these lists after construction?

Reply:
I changed many lists to tuples.